apps/spruce/src/pages/waterfall/BuildRow.tsx

```diff
@@ -128,12 +135,12 @@ const Square = styled(Link)<{ status: string }>`
   cursor: pointer;
   position: relative;
 
-  /* TODO DEVPROD-11368: Render colors for all statuses. Could use background-image property to render icons. */
-  ${({ status }) =>
-    status === TaskStatus.Succeeded
-      ? `background-color: ${green.dark1};`
-      : `background-color: ${gray.light2};
-  `}
+  ${({ status }) => {
+    const icon = statusIconMap?.[status];
```

Comment:
```suggestion
    const icon = statusIconMap[status];
```

Reply:
These are functionally the same since accessing a property that doesn't exist in an object yields `undefined` (no error is thrown)

Reply:
That's right. `icon` may evaluate to undefined but since `statusIconMap` is never undefined, the `?.` is unnecessary. I suggested to make usage with statusColorMap since they are similar.